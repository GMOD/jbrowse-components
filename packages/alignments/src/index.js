import AdapterType from '@gmod/jbrowse-core/pluggableElementTypes/AdapterType'
import DrawerWidgetType from '@gmod/jbrowse-core/pluggableElementTypes/DrawerWidgetType'
import TrackType from '@gmod/jbrowse-core/pluggableElementTypes/TrackType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import { lazy } from 'react'
import {
  configSchema as alignmentsFeatureDetailConfigSchema,
  ReactComponent as AlignmentsFeatureDetailReactComponent,
  stateModel as alignmentsFeatureDetailStateModel,
} from './AlignmentsFeatureDetail'
import {
  configSchemaFactory as alignmentsTrackConfigSchemaFactory,
  modelFactory as alignmentsTrackModelFactory,
} from './AlignmentsTrack'
import {
  AdapterClass as BamAdapterClass,
  configSchema as bamAdapterConfigSchema,
} from './BamAdapter'
import PileupRenderer, {
  configSchema as pileupRendererConfigSchema,
  ReactComponent as PileupRendererReactComponent,
} from './PileupRenderer'

export default class extends Plugin {
  install(pluginManager) {
    pluginManager.addTrackType(() => {
      const configSchema = alignmentsTrackConfigSchemaFactory(pluginManager)
      return new TrackType({
        name: 'AlignmentsTrack',
        configSchema,
        stateModel: alignmentsTrackModelFactory(pluginManager, configSchema),
      })
    })

    pluginManager.addDrawerWidgetType(
      () =>
        new DrawerWidgetType({
          name: 'AlignmentsFeatureDrawerWidget',
          heading: 'Feature Details',
          configSchema: alignmentsFeatureDetailConfigSchema,
          stateModel: alignmentsFeatureDetailStateModel,
          LazyReactComponent: lazy(() => AlignmentsFeatureDetailReactComponent),
        }),
    )

    pluginManager.addAdapterType(
      () =>
        new AdapterType({
          name: 'BamAdapter',
          configSchema: bamAdapterConfigSchema,
          AdapterClass: BamAdapterClass,
        }),
    )

    pluginManager.addRendererType(
      () =>
        new PileupRenderer({
          name: 'PileupRenderer',
          ReactComponent: PileupRendererReactComponent,
          configSchema: pileupRendererConfigSchema,
        }),
    )
  }
}
