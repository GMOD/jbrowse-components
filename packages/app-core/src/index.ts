export { default as App } from './ui/App'
