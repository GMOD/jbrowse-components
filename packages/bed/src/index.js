import AdapterType from '@gmod/jbrowse-core/pluggableElementTypes/AdapterType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import {
  AdapterClass as BigBedAdapterClass,
  configSchema as bigBedAdapterConfigSchema,
} from './BigBedAdapter'

export default class extends Plugin {
  install(pluginManager) {
    pluginManager.addAdapterType(
      () =>
        new AdapterType({
          name: 'BigBedAdapter',
          configSchema: bigBedAdapterConfigSchema,
          AdapterClass: BigBedAdapterClass,
        }),
    )
  }
}
