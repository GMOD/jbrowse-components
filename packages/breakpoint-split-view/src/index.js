import { lazy } from 'react'
import DrawerWidgetType from '@gmod/jbrowse-core/pluggableElementTypes/DrawerWidgetType'
import {
  configSchema as alignmentsFeatureDetailConfigSchema,
  ReactComponent as AlignmentsFeatureDetailReactComponent,
  stateModel as alignmentsFeatureDetailStateModel,
} from './BreakpointAlignmentsFeatureDetail'

export default class BreakpointSplitViewPlugin {
  install(pluginManager) {
    pluginManager.addViewType(() =>
      pluginManager.jbrequire(require('./BreakpointSplitView')),
    )
    pluginManager.addDrawerWidgetType(
      () =>
        new DrawerWidgetType({
          name: 'BreakpointAlignmentsDrawerWidget',
          heading: 'Breakpoint Feature Details',
          configSchema: alignmentsFeatureDetailConfigSchema,
          stateModel: alignmentsFeatureDetailStateModel,
          LazyReactComponent: lazy(() => AlignmentsFeatureDetailReactComponent),
        }),
    )
  }

  configure() {}
}
