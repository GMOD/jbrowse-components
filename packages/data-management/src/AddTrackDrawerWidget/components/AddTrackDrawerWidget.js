import { readConfObject } from '@gmod/jbrowse-core/configuration'
import { getSession } from '@gmod/jbrowse-core/util'
import Button from '@material-ui/core/Button'
import Step from '@material-ui/core/Step'
import StepContent from '@material-ui/core/StepContent'
import StepLabel from '@material-ui/core/StepLabel'
import Stepper from '@material-ui/core/Stepper'
import { makeStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import { observer, PropTypes as MobxPropTypes } from 'mobx-react'
import React, { useState } from 'react'
import ConfirmTrack from './ConfirmTrack'
import TrackSourceSelect from './TrackSourceSelect'

const useStyles = makeStyles(theme => ({
  root: {
    marginTop: theme.spacing(1),
  },
  stepper: {
    backgroundColor: theme.palette.background.default,
  },
  button: {
    marginTop: theme.spacing(1),
    marginRight: theme.spacing(1),
  },
  actionsContainer: {
    marginBottom: theme.spacing(2),
  },
  stepContent: {
    margin: theme.spacing(1),
  },
}))

const steps = ['Enter track data', 'Confirm track type']

function AddTrackDrawerWidget({ model }) {
  const [activeStep, setActiveStep] = useState(0)
  const [trackSource, setTrackSource] = useState('fromFile')
  const [trackData, setTrackData] = useState({ uri: '' })
  const [trackName, setTrackName] = useState('')
  const [trackType, setTrackType] = useState('')
  const [trackAdapter, setTrackAdapter] = useState({})
  const [datasetName, setDatasetName] = useState('')
  const classes = useStyles()

  const session = getSession(model)

  function getStepContent(step) {
    switch (step) {
      case 0:
        return (
          <TrackSourceSelect
            trackSource={trackSource}
            setTrackSource={setTrackSource}
            trackData={trackData}
            setTrackData={setTrackData}
          />
        )
      case 1:
        return (
          <ConfirmTrack
            session={session}
            trackData={trackData}
            trackName={trackName}
            setTrackName={setTrackName}
            trackType={trackType}
            setTrackType={setTrackType}
            trackAdapter={trackAdapter}
            setTrackAdapter={setTrackAdapter}
            datasetName={datasetName}
            setDatasetName={setDatasetName}
          />
        )
      default:
        return <Typography>Unknown step</Typography>
    }
  }

  function handleNext() {
    if (activeStep !== steps.length - 1) {
      setActiveStep(activeStep + 1)
      return
    }
    trackAdapter.features = trackData.config
    const trackConf = session.datasets
      .find(dataset => readConfObject(dataset, 'name') === datasetName)
      .addTrackConf({
        type: trackType,
        name: trackName,
        adapter: trackAdapter,
      })
    if (model.view) {
      model.view.showTrack(trackConf)
    } else {
      session.setSnackbarMessage(
        'Open a new view, or use the track selector in an existing view, to view this track',
      )
    }
    session.hideDrawerWidget(model)
  }

  function handleBack() {
    setActiveStep(activeStep - 1)
  }

  function isNextDisabled() {
    switch (activeStep) {
      case 0:
        return !(
          trackData.uri ||
          trackData.localPath ||
          trackData.blob ||
          trackData.config
        )
      case 1:
        return !(trackName && trackType && trackAdapter.type && datasetName)
      default:
        return true
    }
  }

  return (
    <div className={classes.root}>
      <Stepper
        className={classes.stepper}
        activeStep={activeStep}
        orientation="vertical"
      >
        {steps.map((label, idx) => (
          <Step key={label}>
            <StepLabel>{label}</StepLabel>
            <StepContent>
              {getStepContent(idx)}
              <div className={classes.actionsContainer}>
                <Button
                  disabled={activeStep === 0}
                  onClick={handleBack}
                  className={classes.button}
                >
                  Back
                </Button>
                <Button
                  disabled={isNextDisabled()}
                  variant="contained"
                  color="primary"
                  onClick={handleNext}
                  className={classes.button}
                  data-testid="addTrackNextButton"
                >
                  {activeStep === steps.length - 1 ? 'Add' : 'Next'}
                </Button>
              </div>
            </StepContent>
          </Step>
        ))}
      </Stepper>
    </div>
  )
}

AddTrackDrawerWidget.propTypes = {
  model: MobxPropTypes.observableObject.isRequired,
}

export default observer(AddTrackDrawerWidget)
