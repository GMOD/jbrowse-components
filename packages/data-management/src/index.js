import { lazy } from 'react'
import ConnectionType from '@gmod/jbrowse-core/pluggableElementTypes/ConnectionType'
import DrawerWidgetType from '@gmod/jbrowse-core/pluggableElementTypes/DrawerWidgetType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import {
  configSchema as ucscConfigSchema,
  modelFactory as ucscModelFactory,
} from './ucsc-trackhub'
import {
  reactComponent as AddTrackReactComponent,
  stateModelFactory as AddTrackStateModelFactory,
  configSchema as AddTrackConfigSchema,
} from './AddTrackDrawerWidget'
import {
  reactComponent as AddConnectionReactComponent,
  stateModel as AddConnectionStateModel,
  configSchema as AddConnectionConfigSchema,
} from './AddConnectionDrawerWidget'
import {
  reactComponent as HierarchicalTrackSelectorReactComponent,
  stateModelFactory as HierarchicalTrackSelectorStateModelFactory,
  configSchema as HierarchicalTrackSelectorConfigSchema,
} from './HierarchicalTrackSelectorDrawerWidget'

export default class extends Plugin {
  install(pluginManager) {
    pluginManager.addConnectionType(
      () =>
        new ConnectionType({
          name: 'UCSCTrackHubConnection',
          configSchema: ucscConfigSchema,
          stateModel: ucscModelFactory(pluginManager),
          displayName: 'UCSC Track Hub',
          description: 'A track or assembly hub in the Track Hub format',
          url: '//genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html#Intro',
        }),
    )

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'HierarchicalTrackSelectorDrawerWidget',
        heading: 'Available Tracks',
        configSchema: HierarchicalTrackSelectorConfigSchema,
        stateModel: HierarchicalTrackSelectorStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => HierarchicalTrackSelectorReactComponent),
      })
    })

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'AddTrackDrawerWidget',
        heading: 'Add a track',
        configSchema: AddTrackConfigSchema,
        stateModel: AddTrackStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => AddTrackReactComponent),
      })
    })

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'AddConnectionDrawerWidget',
        heading: 'Add a connection',
        configSchema: AddConnectionConfigSchema,
        stateModel: AddConnectionStateModel,
        LazyReactComponent: lazy(() => AddConnectionReactComponent),
      })
    })
  }
}
