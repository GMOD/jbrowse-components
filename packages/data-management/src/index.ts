import { lazy } from 'react'
import ConnectionType from '@gmod/jbrowse-core/pluggableElementTypes/ConnectionType'
import DrawerWidgetType from '@gmod/jbrowse-core/pluggableElementTypes/DrawerWidgetType'
import Plugin from '@gmod/jbrowse-core/Plugin'
import PluginManager from '@gmod/jbrowse-core/PluginManager'
import {
  SessionWithDrawerWidgets,
  isAbstractMenuManager,
} from '@gmod/jbrowse-core/util'
import NoteAddIcon from '@material-ui/icons/NoteAdd'
import InputIcon from '@material-ui/icons/Input'
import {
  configSchema as ucscConfigSchema,
  modelFactory as ucscModelFactory,
} from './ucsc-trackhub'
import {
  ReactComponent as AddTrackReactComponent,
  stateModelFactory as AddTrackStateModelFactory,
  configSchema as AddTrackConfigSchema,
} from './AddTrackDrawerWidget'
import {
  ReactComponent as AddConnectionReactComponent,
  stateModel as AddConnectionStateModel,
  configSchema as AddConnectionConfigSchema,
} from './AddConnectionDrawerWidget'
import {
  ReactComponent as HierarchicalTrackSelectorReactComponent,
  stateModelFactory as HierarchicalTrackSelectorStateModelFactory,
  configSchema as HierarchicalTrackSelectorConfigSchema,
} from './HierarchicalTrackSelectorDrawerWidget'

export default class extends Plugin {
  install(pluginManager: PluginManager) {
    pluginManager.addConnectionType(
      () =>
        new ConnectionType({
          name: 'UCSCTrackHubConnection',
          configSchema: ucscConfigSchema,
          stateModel: ucscModelFactory(pluginManager),
          displayName: 'UCSC Track Hub',
          description: 'A track or assembly hub in the Track Hub format',
          url: '//genome.ucsc.edu/goldenPath/help/hgTrackHubHelp.html#Intro',
        }),
    )

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'HierarchicalTrackSelectorDrawerWidget',
        heading: 'Available Tracks',
        configSchema: HierarchicalTrackSelectorConfigSchema,
        stateModel: HierarchicalTrackSelectorStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => HierarchicalTrackSelectorReactComponent),
      })
    })

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'AddTrackDrawerWidget',
        heading: 'Add a track',
        configSchema: AddTrackConfigSchema,
        stateModel: AddTrackStateModelFactory(pluginManager),
        LazyReactComponent: lazy(() => AddTrackReactComponent),
      })
    })

    pluginManager.addDrawerWidgetType(() => {
      return new DrawerWidgetType({
        name: 'AddConnectionDrawerWidget',
        heading: 'Add a connection',
        configSchema: AddConnectionConfigSchema,
        stateModel: AddConnectionStateModel,
        LazyReactComponent: lazy(() => AddConnectionReactComponent),
      })
    })
  }

  configure(pluginManager: PluginManager) {
    if (isAbstractMenuManager(pluginManager.rootModel)) {
      pluginManager.rootModel.appendToMenu('File', {
        label: 'Open new track',
        icon: NoteAddIcon,
        onClick: (session: SessionWithDrawerWidgets) => {
          const drawerWidget = session.addDrawerWidget(
            'AddTrackDrawerWidget',
            'addTrackDrawerWidget',
          )
          session.showDrawerWidget(drawerWidget)
        },
      })
      pluginManager.rootModel.appendToMenu('File', {
        label: 'Open new connection',
        icon: InputIcon,
        onClick: (session: SessionWithDrawerWidgets) => {
          const drawerWidget = session.addDrawerWidget(
            'AddConnectionDrawerWidget',
            'addConnectionDrawerWidget',
          )
          session.showDrawerWidget(drawerWidget)
        },
      })
    }
  }
}
