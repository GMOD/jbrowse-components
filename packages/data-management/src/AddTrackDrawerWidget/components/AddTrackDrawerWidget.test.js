import React from 'react'
import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
} from 'react-testing-library'
import { createTestEnv } from '@gmod/jbrowse-web/src/JBrowse'
import AddTrackDrawerWidget from './AddTrackDrawerWidget'

jest.mock('shortid', () => ({ generate: () => 'testid' }))

describe('<AddTrackDrawerWidget />', () => {
  let rootModel
  let model

  beforeAll(async () => {
    ;({ rootModel } = await createTestEnv({
      assemblies: [
        {
          assemblyName: 'volvox',
          sequence: {
            type: 'ReferenceSequenceTrack',
            adapter: {
              type: 'FromConfigAdapter',
              features: [
                {
                  refName: 'ctgA',
                  uniqueId: 'firstId',
                  start: 0,
                  end: 1000,
                  seq:
                    'cattgttgcggagttgaacaACGGCATTAGGAACACTTCCGTCTCtcacttttatacgattatgattggttctttagccttggtttagattggtagtagtagcggcgctaatgctacctgaattgagaactcgagcgggggctaggcaaattctgattcagcctgacttctcttggaaccctgcccataaatcaaagggttagtgcggccaaaacgttggacaacggtattagaagaccaacctgaccaccaaaccgtcaattaaccggtatcttctcggaaacggcggttctctcctagatagcgatctgtggtctcaccatgcaatttaaacaggtgagtaaagattgctacaaatacgagactagctgtcaccagatgctgttcatctgttggctccttggtcgctccgttgtacccaggctactttgaaagagcgcagaatacttagacggtatcgatcatggtagcatagcattctgataacatgtatggagttcgaacatccgtctggggccggacggtccgtttgaggttggttgatctgggtgatagtcagcaagatagacgttagataacaaattaaaggattttaccttagattgcgactagtacaacggtacatcggtgattcgcgctctactagatcacgctatgggtaccataaacaaacggtggaccttctcaagctggttgacgcctcagcaacataggcttcctcctccacgcatctcagcataaaaggcttataaactgcttctttgtgccagagcaactcaattaagcccttggtaccgtgggcacgcattctgtcacggtgaccaactgttcatcctgaatcgccgaatgggactatttggtacaggaatcaagcggatggcactactgcagcttatttacgacggtattcttaaagtttttaagacaatgtatttcatgggtagttcggtttgttttattgctacacaggctcttgtagacgacctacttagcactacgg',
                },
              ],
            },
          },
          tracks: [
            {
              configId: 'i3jUPmrgMOS',
              type: 'FilteringTrack',
              name: 'Filter Test',
              adapter: {
                type: 'FromConfigAdapter',
                features: [
                  {
                    uniqueId: 'one',
                    refName: 'ctgA',
                    start: 100,
                    end: 101,
                    type: 'foo',
                    name: 'Boris',
                    note: 'note for boris',
                  },
                  {
                    uniqueId: 'two',
                    refName: 'ctgA',
                    start: 110,
                    end: 111,
                    type: 'bar',
                    name: 'Theresa',
                    note: 'note for theresa',
                  },
                  {
                    uniqueId: 'three',
                    refName: 'ctgA',
                    start: 120,
                    end: 121,
                    type: 'baz',
                    name: 'Nigel',
                    note: 'note for nigel',
                  },
                  {
                    uniqueId: 'four',
                    refName: 'ctgA',
                    start: 130,
                    end: 131,
                    type: 'quux',
                    name: 'Geoffray',
                    note: 'note for geoffray',
                  },
                ],
              },
              renderer: {
                type: 'SvgFeatureRenderer',
                labels: {},
              },
              filterAttributes: ['type', 'start', 'end'],
            },
          ],
        },
      ],
    }))
    const view = rootModel.addLinearGenomeViewOfAssembly('volvox', {})
    rootModel.addDrawerWidget('AddTrackDrawerWidget', 'addTrackDrawerWidget', {
      view: view.id,
    })
    model = rootModel.drawerWidgets.get('addTrackDrawerWidget')
  })

  afterEach(cleanup)

  it('renders', () => {
    const { container } = render(<AddTrackDrawerWidget model={model} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('adds a track', async () => {
    const { container, getByTestId, getByText, getAllByRole } = render(
      <AddTrackDrawerWidget model={model} />,
    )
    expect(rootModel.configuration.assemblies[0].tracks.length).toBe(1)
    fireEvent.click(getByTestId('addTrackFromConfigRadio'))
    fireEvent.click(getByTestId('addTrackNextButton'))
    fireEvent.change(getByTestId('trackNameInput'), {
      target: { value: 'Test track name' },
    })
    fireEvent.click(getAllByRole('button')[0])
    await waitForElement(() => getByText('BasicTrack'), { container })
    fireEvent.click(getByText('BasicTrack'))
    fireEvent.click(getAllByRole('button')[1])
    await waitForElement(() => getByText('volvox'), { container })
    fireEvent.click(getByText('volvox'))
    fireEvent.click(getByTestId('addTrackNextButton'))
    expect(rootModel.configuration.assemblies[0].tracks.length).toBe(2)
  })
})
