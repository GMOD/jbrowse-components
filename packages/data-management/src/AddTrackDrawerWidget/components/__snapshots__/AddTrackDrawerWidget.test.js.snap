// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTrackDrawerWidget /> renders 1`] = `
<div
  class="makeStyles-root"
>
  <div
    class="MuiPaper-root MuiPaper-elevation0 MuiStepper-root MuiStepper-vertical makeStyles-stepper"
  >
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              1
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
          >
            Enter track data
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root"
        icon="1"
      >
        <div
          class="MuiCollapse-container MuiStepContent-transition MuiCollapse-entered"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner"
            >
              <div
                class="makeStyles-root"
              >
                <fieldset
                  class="MuiFormControl-root"
                >
                  <div
                    aria-label="Data location"
                    class="MuiFormGroup-root"
                    role="radiogroup"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked MuiRadio-checked MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <div
                            class="PrivateRadioButtonIcon-root PrivateRadioButtonIcon-checked"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                              />
                            </svg>
                          </div>
                          <input
                            checked=""
                            class="PrivateSwitchBase-input"
                            data-testid="addTrackFromFileRadio"
                            type="radio"
                            value="fromFile"
                          />
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        Enter data source
                      </span>
                    </label>
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <div
                            class="PrivateRadioButtonIcon-root"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root PrivateRadioButtonIcon-layer"
                              focusable="false"
                              role="presentation"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                              />
                            </svg>
                          </div>
                          <input
                            class="PrivateSwitchBase-input"
                            data-testid="addTrackFromConfigRadio"
                            type="radio"
                            value="fromConfig"
                          />
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        Enter data manually
                      </span>
                    </label>
                  </div>
                </fieldset>
                <div
                  class="MuiPaper-root MuiPaper-elevation1 makeStyles-paper MuiPaper-rounded"
                >
                  <fieldset
                    class="MuiFormControl-root"
                  >
                    <div
                      class="MuiFormGroup-root"
                      role="radiogroup"
                    >
                      <label
                        class="MuiFormControlLabel-root"
                      >
                        <span
                          aria-disabled="false"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-checked MuiRadio-checked MuiIconButton-colorSecondary"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <div
                              class="PrivateRadioButtonIcon-root PrivateRadioButtonIcon-checked"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </div>
                            <input
                              checked=""
                              class="PrivateSwitchBase-input"
                              type="radio"
                              value="uri"
                            />
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                        >
                          URL
                        </span>
                      </label>
                      <label
                        class="MuiFormControlLabel-root MuiFormControlLabel-disabled"
                      >
                        <span
                          aria-disabled="true"
                          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root MuiRadio-root MuiRadio-colorSecondary PrivateSwitchBase-disabled MuiRadio-disabled MuiIconButton-colorSecondary MuiIconButton-disabled MuiButtonBase-disabled"
                          tabindex="-1"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <div
                              class="PrivateRadioButtonIcon-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                                />
                              </svg>
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root PrivateRadioButtonIcon-layer"
                                focusable="false"
                                role="presentation"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                                />
                              </svg>
                            </div>
                            <input
                              class="PrivateSwitchBase-input"
                              disabled=""
                              type="radio"
                              value="localPath"
                            />
                          </span>
                        </span>
                        <span
                          class="MuiTypography-root MuiFormControlLabel-label MuiFormControlLabel-disabled MuiTypography-body1"
                        >
                          Local File
                        </span>
                      </label>
                    </div>
                  </fieldset>
                  <div
                    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                      data-shrink="false"
                    >
                      dataSourceLocation
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        type="text"
                        value=""
                      />
                    </div>
                    
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-actionsContainer"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root makeStyles-button MuiButton-text MuiButton-disabled MuiButtonBase-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Back
                  </span>
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root makeStyles-button MuiButton-contained MuiButton-containedPrimary MuiButton-disabled MuiButtonBase-disabled"
                  data-testid="addTrackNextButton"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiButton-label"
                  >
                    Next
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiStepConnector-root MuiStepConnector-vertical MuiStepConnector-disabled"
    >
      <span
        class="MuiStepConnector-line MuiStepConnector-lineVertical"
      />
    </div>
    <div
      class="MuiStep-root MuiStep-vertical"
    >
      <span
        class="MuiStepLabel-root MuiStepLabel-vertical MuiStepLabel-disabled"
      >
        <span
          class="MuiStepLabel-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiStepIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <circle
              cx="12"
              cy="12"
              r="12"
            />
            <text
              class="MuiStepIcon-text"
              text-anchor="middle"
              x="12"
              y="16"
            >
              2
            </text>
          </svg>
        </span>
        <span
          class="MuiStepLabel-labelContainer"
        >
          <span
            class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
          >
            Confirm track type
          </span>
        </span>
      </span>
      <div
        class="MuiStepContent-root MuiStepContent-last"
        disabled=""
        icon="2"
      />
    </div>
  </div>
</div>
`;
